### PR TITLE
fix hexInt `sprintf` format pattern

### LINKED ISSUE #5462: An error occurred while building
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
Normal building TiFlash.

### 2. What did you expect to see? (Required)
Successful build
### 3. What did you see instead (Required)
```
error: format specifies type 'unsigned long' but the argument has type 'DB::UInt64' (aka 'unsigned long long') [-Werror,-Wformat]
            int print_size = sprintf(reinterpret_cast<char *>(&res_chars[prev_res_offset]), "%lX", number);
```
### 4. What is your TiFlash version? (Required)

```
TiFlash version: master
Compiler version: Apple clang version 13.1.6 (clang-1316.0.21.2.5)
```

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What problem does this PR solve?
Issue Number: close #5462

https://github.com/pingcap/tiflash/pull/5280#issuecomment-1193495227

I should use the `%llX` to print a unsigned uint64 :cry: .

### What is changed and how it works?

Change the format literal from `%lX` to `%llX`

### Release note

```release-note
Fix hexInt `sprintf` format pattern
```

### CODE REVIEW

File: dbms/src/Functions/FunctionsString.cpp [hunk 1]
```diff
@@ -5285,7 +5285,7 @@ class FunctionHexInt : public IFunction
         {
             UInt64 number = col->getUInt(i);
 
-            int print_size = sprintf(reinterpret_cast<char *>(&res_chars[prev_res_offset]), "%lX", number);
+            int print_size = sprintf(reinterpret_cast<char *>(&res_chars[prev_res_offset]), "%llX", number);
```
Comment (fuzhe1989) [inline dbms/src/Functions/FunctionsString.cpp:5288]: use fmt::format instead?

**Review by breezewish — APPROVED** [review]
Verified it works.

**Review by fuzhe1989 — APPROVED** [review]

**Pull request merged**